src/metabase/tiles/api.clj

```diff
@@ -260,34 +251,40 @@
         points (result->points result lat-field-ref lon-field-ref)]
     (tiles-response result zoom points)))
 
-(api.macros/defendpoint :get "/:card-id/:zoom/:x/:y/:lat-field/:lon-field"
+(api.macros/defendpoint :get "/:card-id/:zoom/:x/:y"
   "Generates a single tile image for a saved Card."
-  [{:keys [card-id zoom x y lat-field lon-field]}
-   :- [:merge
-       :api.tiles/route-params
-       [:map
-        [:card-id ms/PositiveInt]]]
-   {:keys [parameters]}
+  [{:keys [card-id zoom x y]}
    :- [:map
-       [:parameters {:optional true} ms/JSONString]]]
-  (let [parameters (json/decode+kw parameters)
-        lat-field  (json/decode+kw lat-field)
-        lon-field  (json/decode+kw lon-field)]
+       [:card-id ms/PositiveInt]
+       [:zoom ms/Int]
+       [:x ms/Int]
+       [:y ms/Int]]
+   {:keys [parameters latField lonField]}
+   :- [:map
+       [:parameters {:optional true} ms/JSONString]
+       [:latField string?]
+       [:lonField string?]]]
+  (let [parameters (when parameters (json/decode+kw parameters))
+        lat-field  (json/decode+kw latField)
```

Comment:
Not your bug, but this will 500 on malformed JSON and we should probably 400 instead... fine for now though.